### DISCUSSION TITLE
how can i get pods status like use 'kubectl get pods', for example:'ContainerCreating','ImagePullBackOff'...

### DISCUSSION THREAD

**Discussion by wanghaininggg** [fabric8io/kubernetes-client#4372] [Q&A]
how can i get pods status like use 'kubectl get pods', for example:'ContainerCreating','ImagePullBackOff'...

**Comment by bhellema** [comment]
How about 
```
pod.getStatus().getContainerStatuses();
// then for each status
status.getState().isWaiting();
// then if waiting
status.getState().getWaiting().getReason();
```

or you can simplify with [PodStatusUtil](https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/PodStatusUtil.java#L174) and get the status then check the status.
